src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs

```diff
@@ -227,12 +236,12 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
                                 Url = filterContext.HttpContext.Request.Url.AbsolutePath,
                                 Tenant = scope.Resolve<ShellSettings>().Name,
                                 StatusCode = response.StatusCode,
-                                Tags = new[] { _invariantCacheKey }.Union(contentItemIds).ToArray()
+                                Tags = new[] { _invariantCacheKey }.Union(contentItemIds).ToArray(),
+                                ETag = etag
                             };
 
                             // Write the rendered item to the cache.
                             var cacheStorageProvider = scope.Resolve<IOutputCacheStorageProvider>();
-                            cacheStorageProvider.Remove(_cacheKey);
```

Comment:
Correct, I assume because the contract is clear, we can safely remove this call, and it's also saving one potential network operation per request (in the case of Redis or FileSystem for instance). The Storage can use the optimized way of updating (Set) a key.